controllers/datadogagent/utils.go

```diff
@@ -670,20 +673,7 @@ func getEnvVarsForAgent(dda *datadoghqv1alpha1.DatadogAgent) ([]corev1.EnvVar, e
 	}
 
 	if spec.ClusterAgent != nil {
```

Comment:
This `if` could be removed since it’s already present inside `envForClusterAgentConnection`.

Reply:
not removed but we can move it under, because `if datadoghqv1alpha1.BoolValue(spec.ClusterAgent.Config.ClusterChecksEnabled)` and lines under it are relying on it